### DISCUSSION TITLE
Is there a way to get all expressions in a template?

### DISCUSSION THREAD

**Discussion by r-j-sharma** [handlebars-lang/handlebars.js#1911] [Q&A]
We would like to get a list of all the expressions in a given template.
Right now we are thinking of using regular expressions. A better way would be to register a special helper that just grabs the expression and then pre compile/compile in a new HandleBar environment. 
Is it possible to register a helper that is called for all expressions?
Is it then possible for this helper to call the default helper?

Thanks!

**Comment by jaylinski** [comment]
Did you already take a look at the Compiler API? It may help you with your problem.

https://github.com/handlebars-lang/handlebars.js/blob/master/docs/compiler-api.md
